ng-dev/utils/yargs.ts

```diff
@@ -0,0 +1,26 @@
+import yargs, {Argv} from 'yargs';
+
+// A function to be called when the command completes.
+type CompletedFn = (err: Error | null) => Promise<void> | void;
+
+/** List of functions to be called upon command completion. */
+const completedFunctions: CompletedFn[] = [];
+
+/** Register a function to be called when the command completes. */
+export function registerCompletedFunction(fn: CompletedFn) {
+  completedFunctions.push(fn);
+}
+
+/**
+ * Run the yargs process, as configured by the supplied function, calling a set of completion
+ * functions after the command completes.
+ */
+export function runParserWithCompletedFunctions(applyConfiguration: (argv: Argv) => Argv) {
+  applyConfiguration(yargs(process.argv.slice(2)))
+    .exitProcess(false)
```

Comment:
From the docs:

>.exitProcess(false) should not be used when [.command()](https://yargs.js.org/docs/#command) is called with a handler returning a promise, as it would lead to a duplicated error message when this promise rejects

is that relevant here

Reply:
Yes, it does in some situations.  Removed.